esphome/components/web_server_base/web_server_base.cpp

```diff
@@ -78,6 +78,11 @@ void OTARequestHandler::handleUpload(AsyncWebServerRequest *request, const Strin
   }
 
   if (final) {
+    // incomplete update! probably tcp reset, bad wifi, etc
+    if (request->contentLength() != this->ota_read_length_) {
+      ESP_LOGI(TAG, "OTA update incomplete, terminating");
+      return;
```

Comment:
This should also call abort to clear out anything
```suggestion
      Update.abort();
      return;
```